apiserver/apiserver.go

```diff
@@ -6,15 +6,21 @@ import (
 	"log"
 	"net/http"
 	"time"
+
+	"github.com/Azure/agentbaker/pkg/agent/toggles"
 )
 
 const (
 	readHeaderTimeoutSeconds = 5
 )
 
+// OptionConfigurator is a function which can configure an Options object.
+type OptionConfigurator func(opts *Options)
+
 // Options holds the options for the api server.
 type Options struct {
-	Addr string
+	Addr    string
+	Toggles *toggles.Toggles
 }
 
 func (o *Options) validate() error {
```

Comment:
this is not exactly related to this PR but just check my own golang sanity:
- line 44 would throw panic if o is nil, 
- shall we check for o == nil here?

Reply:
So in golang you can actually call ptr receivers on nil ptrs - the receiver var will simply be nil within the method itself. We already have a nil check within the validate method, thus calling o.validate won't panic if o is nil, we'll get the error saying options are invalid instead